src/services/BookingService.ts

```diff
@@ -119,13 +121,48 @@ class BookingService {
     return booking;
   }
 
-  public async deleteById(id: number): Promise<void> {
+  public async bookingBelongsToTheUser(
+    bookingId: number, userId: number,
+  ): Promise<boolean> {
+    const repo = this.getRepository();
+    const booking = await repo.findOneBy({ id: bookingId });
+
+    console.log("Booking:");
+    console.log(booking);
+
+    if (Number(booking.user) === userId) {
+      return true;
+    }
+
+    return false;
+  }
+
+  public async deleteById(
+    bookingId: number, userId: number, userRole: string,
+  ): Promise<void> {
+    const isBookingBelongToTheUser = await this.bookingBelongsToTheUser(
+      bookingId, userId,
+    );
+
+    console.log("isBookingBelongToTheUser:");
+    console.log(isBookingBelongToTheUser);
+
+    const isAllowedUser = userRole === UserRole.ADMIN
```

Comment:
Authorization logic should not be done in the service! We have a middleware for that!